lib/maintenance_tasks.rb

```diff
@@ -94,4 +84,29 @@ module MaintenanceTasks
   #
   #  @return [ActiveSupport::Duration] the threshold in seconds after which a task is considered stuck.
   mattr_accessor :stuck_task_duration, default: 5.minutes
+
+  class << self
+    DEPRECATION_MESSAGE = "MaintenanceTasks.error_handler is deprecated and will be removed in the next release. " \
+      "Instead, reports will be sent to the Rails error reporter. Do not set a handler and subscribe" \
+      "to the error reporter instead."
+
+    # no-doc
+    def error_handler
+      deprecator.warn(DEPRECATION_MESSAGE)
+
+      @error_handler
+    end
+
+    # no-doc
+    def error_handler=(proc)
+      deprecator.warn(DEPRECATION_MESSAGE)
+
+      @error_handler = proc
+    end
+
+    # no-doc
+    def deprecator
+      @deprecator ||= ActiveSupport::Deprecation.new
```

Comment:
Should we add the target deprecation horizon + library name? We'd probably need to target the next major version here (3.0)

Reply:
Yes also we want to add it to the application's deprecators when the engine is initialized.